gnucash/gnome/gnc-plugin-page-sx-list.cpp

```diff
@@ -585,9 +585,13 @@ gnc_plugin_page_sx_list_create_widget (GncPluginPage *plugin_page)
         g_object_ref_sink (priv->gdcal);
 
         /* Set number of months from preference, default 12 */
-        gchar *num_months = gnc_prefs_get_string (GNC_PREFS_GROUP_SXED, GNC_PREF_NUM_OF_MONTHS);
-        gnc_dense_cal_set_num_months (priv->gdcal, atoi (num_months));
-        g_free (num_months);
+        auto num_months = gnc_prefs_get_int (GNC_PREFS_GROUP_SXED, GNC_PREF_NUM_OF_MONTHS);
+        if (num_months == 0)
+        {
+            PWARN ("Got invalid value '0' for number of months to display. This suggests a gsettings configuration issue. Continuing with a default value of 12 instead.");
+            num_months = 12;
+        }
+        gnc_dense_cal_set_num_months (priv->gdcal, num_months);
```

Comment:
This fixes [Bug 799470](https://bugs.gnucash.org/show_bug.cgi?id=799470) so please use a bugfix commit message.

But `number-of-months` doesn't show up in dconf in Linux nor org.gnucash.GnuCash.plist on mac, there's no setting for it in Preferences>Scheduled transactions, and changing it in an SX Editor tab doesn't persist if I close the tab and open a new one so I'm a bit mystified about why it works and why it failed for the reporter of that bug.

Reply:
From[Bug 799470 Comment 15] https://bugs.gnucash.org/show_bug.cgi?id=799470#c15):
> The code to store the number of months is only called when you select View->Save layout as default.

That explains it: After Save layout `number-of-months` is in org.gnucash.GnuCash.plist.

Reply:
> This fixes [Bug 799470](https://bugs.gnucash.org/show_bug.cgi?id=799470) so please use a bugfix commit message.
> 

I'm not sure yet this fixes the bug. It does indeed prevent the crash, but as I explain on the bug the crash is the result of not properly being able to access the schema in the first place. We haven't figured out yet why that happens for the bug reporter. If we do need to add code to fix that part as well, I'll use the bugfix commit message for it.

Reply:
OK, no harm in waiting.